src/core/tiledmesh/qgstiledmeshnode.h

```diff
@@ -0,0 +1,178 @@
+/***************************************************************************
+                         qgstiledmeshnode.h
+                         --------------------
+    begin                : June 2023
+    copyright            : (C) 2023 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ******************************************************************
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSTILEDMESHNODE_H
+#define QGSTILEDMESHNODE_H
+
+#include "qgis_core.h"
+#include "qgis.h"
+#include "qgsmatrix4x4.h"
+
+class QgsAbstractTiledMeshNodeBoundingVolume;
+
+/**
+ * \ingroup core
+ * \brief Represents an individual tile node from a tiled mesh data source.
+ *
+ * \since QGIS 3.34
+ */
+class CORE_EXPORT QgsTiledMeshNode
+{
+  public:
+
+    /**
+     * Constructor for an invalid node.
+     *
+     * \see isValid()
+     */
+    QgsTiledMeshNode();
+
+    ~QgsTiledMeshNode();
+
+    //! Copy constructor. Does not copy children or parent!
+    QgsTiledMeshNode( const QgsTiledMeshNode &other );
+    //! Assignment operator. Does not copy children or parent!
+    QgsTiledMeshNode &operator=( const QgsTiledMeshNode &other );
+
+    /**
+     * Returns TRUE if the node is a valid node (i.e. not default constructed).
+     */
+    bool isValid() const { return mIsValid; }
+
+    /**
+     * Adds a \a child to this node.
+     *
+     * Ownership of \a child is transferred to this node, and the parent for \a child will
+     * automatically be set to this node.
+     */
+    void addChild( QgsTiledMeshNode *child SIP_TRANSFER )
+    {
+      child->mParent = this;
+      mChildren.append( child );
+      mIsValid = true;
+    }
+
+    /**
+     * Returns the parent of the node.
+     */
+    QgsTiledMeshNode *parentNode() const { return mParent; }
+
+    /**
+     * Returns the child nodes.
+     */
+    QList< QgsTiledMeshNode * > children() const { return mChildren; }
```

Comment:
with children() one related thing comes to mind - how would we know that if I get no children, whether this means A. this is a leaf node with no further children or B. there may be further children, but we just don't know yet